base_api_connection/models/api_call_mixin.py

```diff
@@ -0,0 +1,68 @@
+# Copyright 2023 Quartile (https://www.quartile.co)
+# License LGPL-3.0 or later (https://www.gnu.org/licenses/lgpl).
+
+import logging
+
+import requests
+
+from odoo import _, models
+from odoo.exceptions import UserError
+
+_logger = logging.getLogger(__name__)
+
+
+class APICallMixin(models.AbstractModel):
+    _name = "api.call.mixin"
+    _description = "API Call Mixin"
+
+    def get_api_key(self, config):
+        if config.token_type:
+            return f"{config.token_type} {config.api_key}"
+        return config.api_key
+
+    def make_api_call(
+        self,
+        code,
+        external_system="generic",
+        endpoint=None,
+        custom_headers=None,
+        params=None,
+        json=None,
+        http_method="get",
+    ):
+        config = self.env["api.config"].search(
+            [("external_system", "=", external_system), ("code", "=", code)], limit=1
+        )
+        if not config:
+            raise UserError(_("API configuration not found."))
+        url = f"{config.base_url}"
+        if endpoint:
+            url = f"{url}/{endpoint}"
+        headers = {"Content-Type": "application/json"}
+        if custom_headers:
+            headers.update(custom_headers)
+        api_key = self.get_api_key(config)
+        headers[config.header_api_key_string] = api_key
+        function = getattr(requests, http_method)
+        kwargs = {"headers": headers}
+        if params:
+            kwargs["params"] = params
+        if json:
```

Comment:
For some rare cases of having to send `{}`, for example.

```suggestion
        if json is not None:
```